src/Processors/Transforms/WindowTransform.cpp

```diff
@@ -2247,6 +2248,12 @@ struct WindowFunctionNonNegativeDerivative final : public StatefulWindowFunction
                             argument_types[ARGUMENT_TIMESTAMP]->getName());
         }
 
+        if (typeid_cast<const DataTypeDateTime64 *>(argument_types[ARGUMENT_TIMESTAMP].get()))
+        {
+            const auto & datetime64_type = assert_cast<const DataTypeDateTime64 &>(*argument_types[ARGUMENT_TIMESTAMP]);
+            ts_scale_multiplier = DecimalUtils::scaleMultiplier<DateTime64>(datetime64_type.getScale());
```

Comment:
Just wondering, if ts_scale_multiplier is set here to 0, we end up in the if branch starting in l. 2308. Is that expected?

Reply:
yes, than the scale as 0 and we can simply treat the value as `Float64`, and no rounding error will occur -- the timestamp is a whole number then